### DISCUSSION TITLE
Pre builds for RPI

### DISCUSSION THREAD

**Discussion by emanuelec84** [djp952/pvr.rtlradio#21] [Q&A]
Can you please add prebuild packages for RPI?
I'm actually using RPI4 and Kodi Leia.

**Comment by djp952** [comment]
Sure, I can cut a build for you at any time to play around with :)  As time allows I've been wasting it on Matrix/Nexus and channel management via Kodi, which isn't even going to be an option for Leia.

Depending on the kernel, you may have play around with installing the "linux-armel", "linux-armhf", or maybe even the "linux-aarch64" build for the Pi 4 (I don't have one, I should really take care of that - LOL) ... here is a link to a freshly minted set of "1.0.0~beta4" .zips for Leia (OneDrive):

https://1drv.ms/u/s!AgEGEEVzGNq-jJwZATfhm6gR6_Qdbg?e=FZcUu3

Note that on Leia, the only way to import channels is via the **Settings / PVR & Live TV / Client specific / Client specific settings / Import channel data** operation I describe (poorly) here: https://github.com/djp952/pvr.rtlradio/blob/Leia/doc/importexport.md

This thing is far from perfect, maybe even barely workable, and I have no documentation yet for either Linux or Windows on how to get the RTL-SDR device to be seen by Kodi, but I'm guessing you already know how to make that happen (blacklisting the DVB specific driver)?  If not, our collective experience will illustrate what documentation is needed, right?

Have at it!  I've been mired down in both my "real work" and trying to get Kodi and the addon into a place to manage channels without the .JSON that progress is so slow it probably seems non-existent :(

Also make sure to have a look at the currently document Issues (https://github.com/djp952/pvr.rtlradio/issues), and PLEASE add more for me as you find them, OK?  There are some things that I am aware of that can't be fixed outside of Kodi, but whatever you find/see/feel about this I really want to know.

Oh, one more thing, if you are intending to use **rtl_tcp** for this, I strongly suggest using my rtl_tcp implementation (inside the linked rtlsdr-toolkit-{platform}-{version}.zip files for the device hosting the RTL-SDR dongle) as the osmocom version kind of sucks and can't handle something like Kodi that needs to change channels in a reasonable timeframe.  I have been unable to get my proposed changes accepted by osmocom to date, but I've been using my rtl_tcp (armhf) on a WD EX4100 NAS for months and it works pretty much perfectly :)

Looking forward to your feedback/criticisms.  No feedback is "bad", just positive/negative, which are both necessary.

**Reply by emanuelec84** [reply]
Thank you so much for your kindness!
I'll try and let you know how it goes.
I'm already aware of the process of installing the sdr stick and it's working pretty fine. What I'd really like is to have a workable addon for Kodi.
I pretty busy too with "real work" and with kids.
I'd like to try Matrix on raspberry pi 4 but actually the only way is to move to openElec or build it from sources.
Both solution are not feasable at the moment.
First one would imply a loss of functionality, second one would problably required plenty of time I don't have now.
Did you manage to make Matrix run on raspi Os (raspbian)?
I took a look and import channel data doc and I noticed it's the same one of Matrix. What other way to import data on Matrix?

Are you aware of a similar add on for DAB channels?

BR,

Emanuele

**Reply by djp952** [reply]
I don't actually use the Raspberry Pi OS on anything ATM, I wasn't even aware Matrix was an issue for it.  The upgrades from Leia to Matrix have been pretty brutal, that Python switchover really put the distros in a bad place.  OSMC is catching up and I think LibreELEC is getting close but I think both are also lagging for the Pi 4.

On Matrix, the *idea* was to be able to use Kodi's built-in Channel Manager, and it kinda/sorta works, it's just a bit wonky.  I've been trying to clean up the wonkiness with PRs, but I think Nexus (Kodi 20) may have already diverged too much from Matrix in the PVR code for them to let me backport anything into Matrix.  The reason this isn't available in Leia is because that version of the PVR addon interface wouldn't allow for custom dialog boxes.

I don't personally know of anything for DAB, but being in the US we don't have anything like that to even much around with.  I think in theory the process would be the same, find something that does DAB via the RTL-SDR dongle and just adapt it.  I barely understand how the actual demodulator works here, but to quote Han Solo "I've added some special modifications myself" :)  I know enough to be dangerous, basically.

**Comment by djp952** [comment]
So ... since the last part of this discussion I've killed Leia completely due to the systemic Kodi issues that will never be fixed (sorry), and finally got to a "1.0.0" for Matrix (https://github.com/djp952/pvr.rtlradio/releases/tag/matrix-1.0.0).  The wiki/documentation is very much still a work in progress, but it's out there.  I'll work on the documentation as time allows over the next few days.

As mentioned, not all my Kodi fixes were backported to Matrix, but it's "serviceable".  To my knowledge the only "kit" distro for most devices that's up to Matrix is OSMC.  LibreELEC claims to be close for the RPI4, but I don't have one so I really can't try it :(

I'm sorry that I had to kill Leia, but it just wasn't working out.  I wish Kodi would allow fixes/backports for prior versions but they simply do not.  When the time comes I hope to hear your thoughts/experiences with the addon on Matrix.  For what it's worth, I've updated the "Issues" that I am aware of (https://github.com/djp952/pvr.rtlradio/issues) to show that things are still far from perfect.  But, "1.0.0" is out there, for better or for worse :)
